src/dwarf.cpp

```diff
@@ -660,11 +669,36 @@ std::string dwarf::implementation::qualified_symbol_name(
     return result;
 }
 
+/**************************************************************************************************/
+// incoming path comes from e.g., `file_decl` attribute, or one of the `_file_decl` entries.
+// Given what we know of the Mach-O file, attempt to canonicalize the path, making it absolute
+// if it is relative, and we have a `_cu_compilation_directory`. This is a relatively expensive
+// operation (possibly requiring dynamic memory allocations) so should not be used during the
+// typical processing path.
+pool_string dwarf::implementation::make_path_canonical(pool_string candidate) {
+    std::filesystem::path path(candidate.view());
+
+    if (path.is_absolute()) {
+        return candidate;
+    }
+
+    // Hopefully we got `_cu_compilation_directory` from the compilation unit die
+    // processed prior to this one.
+    if (!_cu_compilation_directory) {
+        return candidate;
+    }
+
+    path = weakly_canonical(_cu_compilation_directory.view() / std::move(path));
+
+    return empool(std::move(path).string());
```

Comment:
isn't path - after std::move - invalid here?

Reply:
It's not - a `std::move` is just a cast - nothing actually _moves_ with `std::move`. Rather, the cast turns the variable into an r-value reference, which permits the compiler to use additional functions that take r-value refs as parameters. In _this_ case, what I'm trying to do is tell the compiler, "hey, this path is about to die, so if there's a way the string can be made by gutting the path and putting it into the string, go for it, save some copies, and I pinky-swear not to mess with `path` after this expression."

Reply:
Missed the assignment - looks good!